### ISSUE TITLE
Update which Go Modules have a version() func

### ISSUE THREAD

**Issue by eddyleelin** [open-telemetry/opentelemetry-go#2143]
### Problem Statement

Currently, a single [`version.go`](https://github.com/open-telemetry/opentelemetry-go/blob/main/version.go) file lives in the repo root directory, containing a `Version()` func that returns a hard-coded string with the version of the `module go.opentelemetry.io/otel` as defined in the `go.mod` in the same directory.

The main question is whether it would help maintain consistency and help users to have a `version.go` file live alongside ALL `go.mod` files, containing the same type of `Version()` function, rather than just have the single one at the root. 

### Proposed Solution

By having a `version.go` file define the `Version()` func for every module within the repo, it would be easy for users to call `mod.Version()` while using any Go Module from the repo. It would not require much additional work to maintain, as the Releaser app (described in Additional Context) could automatically search the directories holding `go.mod` files for the corresponding `version.go` file.

#### Alternatives

Alternatively, it may not be necessary to support the `Version()` func for every module. In this case, the Releaser app could still search for any `version.go` files in the repo and only modify existing ones.

### Additional Context

The Releaser app being built in the OpenTelemetry-Go-Build-Tools repo ([PR here](https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/8)) has already implemented the functionality to update any old version numbers in `go.mod` files to new version numbers as specified in the `versions.yaml` file. However, there is an unimplemented function stub that could update all `version.go` files in the repo.

**Comment by MrAlias** [comment]
> The main question is whether it would help [...] users to have a `version.go` file live alongside ALL `go.mod` files ...

Agreed. Can you add user use-cases where it would be helpful and not something that could be solved with the current setup?

**Comment by Aneurysm9** [comment]
This was discussed during the SIG meeting today and we concluded that each signal SDK package should have a `version.go` that has `func version() string` that returns the version of the module containing the SDK package.  At this time that would be `sdk/trace` and `sdk/metric`.  

We did not have any readily available use case for exporting the version information, so the function should be unexported until such time as requirements dictate otherwise.

**Comment by deepto98** [comment]
Can I pick this up? @MrAlias Can you point me to the files I'd need to look at?

**Comment by ChillOrb** [comment]
Hello @eddyleelin , please review #3677

**Comment by flc1125** [comment]
@open-telemetry/go-approvers This issue seems to have been resolved, but has not been closed.
